src/PrestaShopBundle/Form/Admin/Sell/Supplier/SupplierType.php

```diff
@@ -0,0 +1,354 @@
+        }
+
+        $builder
+            ->add('name', TextType::class, [
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->translator->trans(
+                            'This field cannot be empty', [], 'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new Length([
+                        'max' => 64,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters',
+                            ['%limit%' => 64],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                    new TypedRegex([
+                        'type' => 'catalog_name',
+                    ]),
+                ],
+            ])
+            ->add('description', TranslatableType::class, [
+                'type' => TextareaType::class,
+                'required' => false,
+                'options' => [
+                    'constraints' => [
+                        new CleanHtml([
+                            'message' => $this->translator->trans(
+                                '%s is invalid.',
+                                [],
+                                'Admin.Notifications.Error'
+                            ),
+                        ]),
+                    ],
+                ],
+            ])
+            ->add('phone', TextType::class, [
+                'required' => false,
+                'constraints' => [
+                    new TypedRegex([
+                        'type' => 'phone_number',
+                    ]),
+                    new Length([
+                        'max' => 32,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters',
+                            ['%limit%' => 32],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                ],
+            ])
+            ->add('mobile_phone', TextType::class, [
+                'required' => false,
+                'constraints' => [
+                    new TypedRegex([
+                        'type' => 'phone_number',
+                    ]),
+                    new Length([
+                        'max' => 32,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters',
+                            ['%limit%' => 32],
+                            'Admin.Notifications.Error'
+                        ),
+                    ]),
+                ],
+            ])
+            ->add('address', TextType::class, [
+                'constraints' => [
+                    new TypedRegex([
+                        'type' => 'address',
```

Comment:
Wondering whether we should have used constants for TypedRegexValidator 🤔

Reply:
Cc @sarjon I believe we had this discussion with another PR

Reply:
🆙

Reply:
Hmmm, I don't recall having discussion about constants, but I think that would make sense. :+1: